### DISCUSSION TITLE
Climate control

### DISCUSSION THREAD

**Discussion by Jesus-M** [Hyundai-Kia-Connect/hyundai_kia_connect_api#656] [Ideas]
The `start_climate` func uses the `ClimateRequestOptions` class as parameter, with the variables:

```
    set_temp: float = None
    duration: int = None
    defrost: bool = None
    climate: bool = None
    heating: int = None
    front_left_seat: int = None
    front_right_seat: int = None
    rear_left_seat: int = None
    rear_right_seat: int = None
```

(by the way, what exactly is `heating`? and `climate`? what do they control?)

But there are other parameters not controlled/updated by this function, like:

```
    steering_wheel_heater_is_on: bool = None
    back_window_heater_is_on: bool = None
    side_mirror_heater_is_on: bool = None
```

Is there any function I can use to enable/disable these features?  Could these be added to the `start_climate` func (to the `ClimateRequestOptions` class probably)?

I am trying to automate the start of the climate, in the same way I do it with the **Bluelink** app, where I just hit the 'start' and all these options are enabled.

Thanls

**Comment by cdnninja** [comment]
Climate enables or disables hvac.   Almost always should be enabled. 

Heating is an integer that each number is a combo of things being enabled or disabled.  Like steering wheel plus front window.   I think constants show that.

**Reply by Jesus-M** [reply]
Oh, I see, the `heating` value should include what I was asking for, right?

So I just need to look for the right values in the constants table, isn't it?

Thanks!

**Reply by cdnninja** [reply]
Yes essentially.   So the Enums are here: https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/blob/b30ab21dc680a7a514c02dfd73adf6d6728089a3/hyundai_kia_connect_api/const.py#L57-L65

So if you want just rear window send a 2.  Some playing may be needed to know what works for your car. 

Now we have found that this seems to start to vary by car and region.  Very difficult to get an accurate list.   Similar approach exists to heated seat status.

**Reply by Jesus-M** [reply]
Thanks a lot!

And when I figure out the values for my car/region, I will share, in case that helps someone else.

Thanks again

**Reply by SouthernGecko** [reply]
Hi,
If for example the front_left_seat parameter is set to 6 in the ClimateRequestOptions, should that start the seat heating in the car?
(It doesn't work for me (EU region) and it doesn't look like this is read in the code, unlike the "heating" which works).

**Reply by cdnninja** [reply]
> Hi,
> 
> If for example the front_left_seat parameter is set to 6 in the ClimateRequestOptions, should that start the seat heating in the car?
> 
> (It doesn't work for me (EU region) and it doesn't look like this is read in the code, unlike the "heating" which works).
> 
> 
> 
> 


Depends does your car support levels from within the native app?  Based on this it will be either 1 or a level.  Depending on the car. 

https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/blob/master/hyundai_kia_connect_api/const.py#L44-L55


EDIT: good catch.  Seats aren't implemented for EU. if someone sniffs the traffic probably not hard to add.

**Reply by Jesus-M** [reply]
I didn't figure it out for my car (Hyundai Kona) in EU. The number of API requests is very limited, and I need at least three queries for every test, makes it very long to try.

**Reply by SouthernGecko** [reply]
> Depends does your car support levels from within the native app? Based on this it will be either 1 or a level. Depending on the car.
> EDIT: good catch. Seats aren't implemented for EU. if someone sniffs the traffic probably not hard to add.

I don't see it in the native app so I don't know if it's possible.
